api/detail.mjs

```diff
@@ -130,28 +132,139 @@ const parseDoorDashStore = (storeData) => {
  */
-const detailStore = async ({ service, storeId }) => {
+const detailStore = async (serviceIds) => {
   const services = {
-    postmates: Postmates,
-    grubhub: Grubhub,
-    doordash: Doordash,
+    postmates: { instance: Postmates, parser: parsePostmatesStore },
+    grubhub: { instance: Grubhub, parser: parseGrubhubStore },
+    doordash: { instance: Doordash, parser: parseDoorDashStore },
   };
-  const serviceInstance = new services[service]();
-  const store = await serviceInstance.getStore(storeId);
-
-  switch (service) {
-    case "postmates":
-      return parsePostmatesStore(store);
-    case "grubhub":
-      return parseGrubhubStore(store);
-    case "doordash":
-      return parseDoorDashStore(store);
-  }
+
+  return Promise.all(
+    serviceIds.reduce((accServices, { service, id }) => {
+      if (id) {
+        accServices.push(
+          new services[service].instance().getStore(id).then((store) => ({
+            service: service,
+            ...services[service].parser(store),
+          }))
+        );
+      }
+      return accServices;
+    }, [])
+  ).then((serviceStores) => {
+    let menu = {};
+
+    const defaultService = serviceStores[0];
+
+    for (const { category, categoryId, items } of defaultService.menu) {
+      addCategoryToMenu({
+        category: { id: categoryId, name: category },
+        menu: menu,
+        items: {},
+        service: defaultService.service,
+      });
+      for (const item of items) {
+        addItemToMenu({
+          categoryItems: menu[category].items,
+          service: defaultService.service,
+          item: item,
+        });
+      }
+    }
+
+    serviceStores.shift();
+
+    // merging menu items of the same category
+    for (const serviceStore of serviceStores) {
+      for (const { categoryId, category, items } of serviceStore.menu) {
+        if (menu[category]) {
+          menu[category].categoryIds[serviceStore.service] = categoryId;
+          for (const item of items) {
+            if (menu[category].items[item.name]) {
+              console.log(item.name);
+              menu[category].items[item.name].prices[serviceStore.service] =
+                item.price;
+              menu[category].items[item.name].ids[serviceStore.service] =
+                item.id;
+            }
+            // if item does not already exist, add it
+            else {
+              addItemToMenu({
+                categoryItems: menu[category].items,
+                service: serviceStore.service,
+                item: item,
+              });
+            }
+          }
+          // if category does not already exist, add it
+        } else {
+          addCategoryToMenu({
+            category: { id: categoryId, name: category },
+            menu: menu,
+            items: {},
+            service: serviceStore.service,
+          });
+        }
+      }
+    }
+
+    // removing application specific categories
+    // postmates
+    delete menu["Picked for you"];
```

Comment:
Just a thought, IIRC one of our use cases was finding popular items at a restaurant? Maybe we can repurpose this later idk.

Reply:
Yeah, I was thinking about that as well, this can likely be used for that as well. We would need to look at all of those popular items and see if we can find matching pairs of those items in the other services to get the associated ids.